pydantic/_internal/_mock_val_ser.py

```diff
@@ -108,8 +109,48 @@ def rebuild(self) -> ValSer | None:
         return None
 
 
+def set_type_adapter_mocks(adapter: TypeAdapter, type_repr: str) -> None:
+    """Set `core_schema`, `validator` and `serializer` to mock core types on a type adapter instance.
+
+    Args:
+        adapter: The type adapter instance to set the mocks on
+        type_repr: Name of the type used in the adapter, used in error messages
+    """
+    undefined_type_error_message = (
+        f'`TypeAdapter[{type_repr}]` is not fully defined; you should define `{type_repr}` and all referenced types,'
+        f' then call `.rebuild()` on the instance.'
+    )
+
+    def attempt_rebuild_fn(attr_fn: Callable[[TypeAdapter], T]) -> Callable[[], T | None]:
+        def handler() -> T | None:
+            if adapter.rebuild(_parent_namespace_depth=5) is not False:
+                return attr_fn(adapter)
+            else:
+                return None
+
+        return handler
+
+    adapter.core_schema = MockCoreSchema(  # type: ignore[assignment]
```

Comment:
Could we make use of pyright specific ignore comments? (`#pyright: ignore[reportAssignmentType]`)